### ISSUE TITLE
Account Overdraft can be enabled or disabled

### ISSUE THREAD

**Issue by SilasReinagel** [SOOlutions/HomeTaskApp#4]
Account should have Overdrafts disabled by default.
If Overdrafting is disabled, the Account Balance should never drop below 0. 
If Overdrafting is enabled, the Account Balance can be any value. 
A user should be able to enable Overdrafting.
A user should not be able to disable Overdrafting while he has a negative balance.

**Comment by vivekimsit** [comment]
This counts as business case or application logic, right?

**Comment by SilasReinagel** [comment]
@vivekimsit What is "application logic"? Does such a thing exist?

**Comment by vivekimsit** [comment]
From my experience we try to separate object's core behaviour,  are these core behaviours of the objects?

**Comment by SilasReinagel** [comment]
Depends on how it is architected. If an Object makes his own decisions, then it would seem to be a core trait of the Account himself. 

Wouldn't separating out various behaviors of an Account and putting them elsewhere de-objectify the Account?

**Comment by vivekimsit** [comment]
Yes, that was more of a question than answer from me, understanding this thing is most important for me because I get confused where to draw that boundary i.e. how much behaviour should go into an object?

**Comment by goqp** [comment]
This is what i meant about abstracting the architecture from the object.
Right now, the behavior that falls within the domain of an object (account type stuff) needs to be done by some kind of an account. If the behavior is less core than it should go into a decorator.

**Comment by goqp** [comment]
The architecture / business logic is inside the ticket and needs to come out.

Ticket a. Account
1. Account should be able to add and remove policies (for example, can overdraft).
2. Only authorized users can add a given policy. 

Ticket b. Business logic
1. Account should have Overdrafts disabled by default.
2. If Overdrafting is disabled, the Account Balance should never drop below 0.
3. If Overdrafting is enabled, the Account Balance can be any value.
4. A user should be able to enable Overdrafting.
5. A user should not be able to disable Overdrafting while he has a negative balance.

**Comment by SilasReinagel** [comment]
This can be modelled as Account having Policies. You can break this into two tickets if needed.
